include/envoy/http/filter.h

```diff
@@ -224,6 +224,12 @@ class StreamFilterBase {
   virtual ~StreamFilterBase() {}
 
   /**
+   * This routine is called on filter creation, allowing filters to subscribe to watermark events
+   * on the downstream stream and downstream connection.
+   */
+  virtual void provideWatermarkCallbacks(Http::DownstreamWatermarkProvider& provider) PURE;
```

Comment:
Just to clarify my understanding (and maybe a comment clarification opportunity), the reason we do this dance with the `DownstreamWatermarkProvider`, rather than just making the above/below methods part of the filter interface, is to avoid having every filter have its above/below callbacks invoked on every watermark event? I.e. to narrow the scope of the watermark crossing event firing.